src/philips_airctrl/discovery.py

```
@@ -0,0 +1,157 @@
+    async def scan_ip(self, ip: str) -> DeviceInfo | None:
+        """Scan a single IP address for a Philips air purifier."""
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.settimeout(1.0)
+            try:
+                result = sock.connect_ex((ip, 5683))
+                sock.close()
+
+                if result != 0:
+                    return None
+            except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)